itest/rust/src/builtin_tests/string/gstring_test.rs

```diff
@@ -95,3 +95,25 @@ fn string_hash() {
     .collect();
     assert_eq!(set.len(), 5);
 }
+
+#[itest]
+#[cfg(feature = "serde")]
+fn serde_roundtrip() {
+    #[derive(serde::Deserialize, serde::Serialize, PartialEq, Debug)]
+    struct P {
+        name: GString,
+    }
+    let value = P {
+        name: GString::from("hello world"),
+    };
+    let expected_json = "{\"name\":\"hello world\"}";
+
+    let json: String = serde_json::to_string(&value).unwrap();
+    let back: P = serde_json::from_str(json.as_str()).unwrap();
+
+    assert_eq!(back.name, value.name, "serde round-trip changes value");
+    assert_eq!(
+        json, expected_json,
+        "value does not conform to expected JSON"
+    );
+}
```

Comment:
This seems to test `serde`'s own composability inside structs more than our `GString` serialization. 
I think we could directly serialize/deserialize the `GString`, without `P` -- especially since the array test already composes them.